src/Nethermind/Nethermind.Mev/Source/BundleSortedPool.cs

```diff
@@ -0,0 +1,47 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Core.Collections;
+using Nethermind.Logging;
+using Nethermind.Mev.Data;
+using Nethermind.TxPool.Collections;
+
+namespace Nethermind.Mev.Source
+{
+    public class BundleSortedPool : DistinctValueSortedPool<MevBundle, MevBundle, long> 
+    {
```

Comment:
is it needed?

Reply:
It gives us very good, prod proven eviction mechanism as well as grouping and sorting.